lib/ansible/config/manager.py

```diff
@@ -101,7 +101,10 @@ def ensure_type(value, value_type, origin=None, origin_ftype=None):
             value = boolean(value, strict=False)
 
         elif value_type in ('integer', 'int'):
-            value = int(value)
+            if isinstance(value, float) or isinstance(value, string_types) and not value.lstrip('-').isnumeric():
```

Comment:
Wouldn't `str.isdigit()` be more appropriate here?
https://docs.python.org/3/library/stdtypes.html#str.isnumeric says that Unicode chars like https://www.compart.com/en/unicode/U+2155 would also match, which is probably not what we want.

Reply:
yes